### DISCUSSION TITLE
Why is there no .cbc file in the output file of my model

### DISCUSSION THREAD

**Discussion by Alvin-prc** [modflowpy/flopy#1988] [Q&A] [answered]
#1. 
name_flow = 'SM_flow'  # 
workspace = './Model_files'  # 
mf = flopy.modflow.Modflow(modelname=name_flow, version="mf2005",
                           exe_name="mf2005dbl.exe", model_ws=workspace)
###################################################################################################
#2. 
Lx = 800.0  # 
Ly = 800.0  # 
ztop = 1.0  #
zbot = 0.0  # 
nlay = 1  # 
nrow = 80  
ncol = 80  # 
delr = Lx / ncol  #
delc = Ly / nrow  # 
botm = np.linspace(ztop, zbot, nlay + 1)  #
nper = 2  # 
perlen = [0.05, 5.0]  # 
nstp = [1, 100]  # 
steady = [True, False]  # 
#Unit System
itmuni = 4  # time units (4=days, 1=seconds)
lenuni = 2  # lenght units (2=meters, 3=centimeters)
dis = flopy.modflow.ModflowDis(mf, nlay, nrow, ncol, delr=delr,
                               delc=delc, top=ztop, botm=botm[1:],
                               nper=nper, perlen=perlen, nstp=nstp, steady=steady,
                               itmuni=itmuni, lenuni=lenuni)
###################################################################################################
#3. 
ibound = np.ones((nlay, nrow, ncol), dtype=np.int32)
ibound[:, :, 0] = -1  #
strt = np.zeros((nlay, nrow, ncol), dtype=np.float32)  # 

bas = flopy.modflow.ModflowBas(mf, ibound=ibound, strt=strt)
##################################################################################################
#4. LPF
laytyp = 0  # 
vka = 10  # 
#note 
hk = np.exp(np.load('../data/real_lnK.npy'))
#
lpf = flopy.modflow.ModflowLpf(mf, laytyp=0, hk=hk, layvka=1, vka=vka)
###################################################################################################
#5. CHD 
shead = 0.0  # 
ehead = 0.0  # 
bound_sp1 = []
for il in range(nlay):
    for ir in range(nrow):
        bound_sp1.append([il, ir, 0, shead, ehead])

stress_period_data = {0: bound_sp1}
#Create the flopy chd object
chd = flopy.modflow.ModflowChd(mf, stress_period_data=stress_period_data)
#################################################################################################
#6. 
pumping_rate1 = -20.0
pumping_rate2 = 0.0
wel_sp1 = []
for ir in range(nrow):
    wel_sp1.append([0, ir, ncol - 1, pumping_rate1])

wel_sp2 = []
for ir in range(nrow):
    wel_sp2.append([0, ir, ncol - 1, pumping_rate2])

stress_period_data = {0: wel_sp1, 1: wel_sp2}
wel = flopy.modflow.ModflowWel(mf, stress_period_data=stress_period_data)

###################################################################################################
#8. OC
#
stress_period_data = {}
for kper in range(nper):
    for kstp in range(nstp[kper]):
        stress_period_data[(kper, kstp)] = [
            "save head",  # 
            "save drawdown",  # 
            "save budget",  # 
            "print head",
            "print budget",
        ]
oc = flopy.modflow.ModflowOc(mf, stress_period_data=stress_period_data, compact=True)
###################################################################################################
#9. PCG 
pcg = flopy.modflow.ModflowPcg(mf)

#
mf.write_input()

#
success, mfoutput = mf.run_model()
if not success:
    raise Exception('MODFLOW did not terminate normally!')

####################################################################################################
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2021.3.1\plugins\python\helpers\pydev\pydevd.py", line 1500, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2021.3.1\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:\Project_Work\My-Own-Work\23.10.18\real-situation\real_flopy.py", line 144, in <module>
    budgobj = bf.CellBudgetFile(os.path.join(workspace, name_flow+'.cbc'))
  File "C:\ProgramData\Anaconda3\lib\site-packages\flopy\utils\binaryfile.py", line 768, in __init__
    self.file = open(self.filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\Project_Work\\My-Own-Work\\23.10.18\\real-situation\\Model_files\\SM_flow.cbc'

**Comment by Alvin-prc** [comment]
please!

**Comment by geoscience-coding** [comment]
Because you like don’t have a setting to save budgets out for a stress period.  Sent from my iPhone516-637-2534On Oct 21, 2023, at 12:41 AM, Meng ***@***.***> wrote:﻿
please!

**Reply by Alvin-prc** [reply]
I set "save budget" in output control, is something wrong?

8. OC
stress_period_data = {}
for kper in range(nper):
for kstp in range(nstp[kper]):
stress_period_data[(kper, kstp)] = [
"save head", #
"save drawdown", #
"save budget", #
"print head",
"print budget",
]
oc = flopy.modflow.ModflowOc(mf, stress_period_data=stress_period_data, compact=True)

**Comment by wpbonelli** [comment, marked as answer]
@Alvin-prc you either need to set nonzero `ipakcb` on the packages for which you would like budget output saved, or you can call `oc.reset_budgetunit()` to enable it for all relevant packages.

**Comment by geoscience-coding** [comment]
I haven’t tried your code, but looks like you need to save budgets all or for specific stress periods.  See example flopy.mf6.ModflowGwfoc(            gwf,
            head_filerecord=head_filerecord,
            budget_filerecord=budget_filerecord,
            saverecord=[("HEAD", "ALL"), ("BUDGET", "ALL")],Sent from my iPhone516-637-2534On Oct 22, 2023, at 8:22 AM, wpbonelli ***@***.***> wrote:﻿
@Alvin-prc you either need to set nonzero ipakcb on the packages for which you would like budget output saved, or you can call oc.reset_budgetunit() to enable it for all relevant packages.
